repl/src/main/scala/ammonite/repl/Storage.scala

```diff
@@ -57,6 +58,57 @@ object Storage{
       }
     }
 
+    def compileCacheSave(tag: String, classFiles: Traversable[(String, Array[Byte])], imports: Seq[ImportData]): Unit = {
+      val cacheDir = new File(dir + s"/compileCache/$tag")
+      if(!cacheDir.exists){
+        cacheDir.mkdirs()
+        val metadata = upickle.default.write(imports)
+        val metadataFw = new FileWriter(cacheDir + "/metadata.json")
```

Comment:
Since you're adding a second case of this, let's break out all this `FileWriter` logic into a helper function so `ivyCache` below can re-use it